compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -52,6 +53,8 @@ object PatternMatcher {
   /** Minimal number of cases to emit a switch */
   final val MinSwitchCases = 4
 
+  object TrustedTypeTestKey extends Key[Unit]
```

Comment:
Make it a `StickyKey` to avoid it being tripped accidentally in a tree copier. Also, we usually just use a val for this.
```
val TrustedTypeTestKey: Key[Unit] = new StickyKey[Unit]
```